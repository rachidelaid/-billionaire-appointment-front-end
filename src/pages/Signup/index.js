import React from 'react';
import { Link } from 'react-router-dom';
import style from './style.module.css';

const Signup = () => (
  <div className={style.container}>
    <div className={style.wrapper}>
      <div className={style.card}>
        <h1>Signup</h1>

        <input type="text" placeholder="Name" />
        <input type="text" placeholder="Username" />
        <input type="email" placeholder="Email" />
        <input type="password" placeholder="Password" />
        <input type="password" placeholder="Confirm Password" />

        <button type="button" className={style.btn}>Sign Up</button>
      </div>
      <small className={style.option}>
        Already have an account?
        {' '}
        <Link to="/login">Sign In</Link>
      </small>
    </div>
  </div>
);

export default Signup;
