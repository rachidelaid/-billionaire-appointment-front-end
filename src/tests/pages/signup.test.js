import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Signup from '../../pages/Signup';
import Wrapper from '../helpers/test_wrapper';

describe('Signup Page Snapchat', () => {
  const page = render(<Wrapper><Signup /></Wrapper>);

  it('should render the signup page', () => {
    expect(page).toMatchSnapshot();
  });
});

describe('Signup Page Elements', () => {
  beforeEach(() => {
    render(<Wrapper><Signup /></Wrapper>);
  });

  it('should show the Name input', async () => {
    const nameInput = await screen.findByPlaceholderText('Name');
    expect(nameInput).toBeInTheDocument();
  });

  it('should show the Name input', async () => {
    const usernameInput = await screen.findByPlaceholderText('Name');
    expect(usernameInput).toBeInTheDocument();
  });

  it('should show the Email input', async () => {
    const emailInput = await screen.findByPlaceholderText('Email');
    expect(emailInput).toBeInTheDocument();
  });

  it('should show the Password input', async () => {
    const passwordInput = await screen.findByPlaceholderText('Password');
    expect(passwordInput).toBeInTheDocument();
  });

  it('should show the Confirm Password input', async () => {
    const confirmPasswordInput = await screen.findByPlaceholderText('Confirm Password');
    expect(confirmPasswordInput).toBeInTheDocument();
  });

  it('should show the sign-up button', async () => {
    const signupButton = await screen.findByText('Sign Up');
    expect(signupButton).toBeInTheDocument();
  });
});

describe('Signup Page Errors Handler', () => {
  it('should show the wrong email error', async () => {
    render(<Wrapper><Signup /></Wrapper>);

    const nameInput = await screen.findByPlaceholderText('Name');
    const usernameInput = await screen.findByPlaceholderText('Username');
    const emailInput = await screen.findByPlaceholderText('Email');
    const passwordInput = await screen.findByPlaceholderText('Password');
    const confirmPasswordInput = await screen.findByPlaceholderText('Confirm Password');

    fireEvent.change(nameInput, { target: { value: 'test' } });
    fireEvent.change(usernameInput, { target: { value: 'test' } });
    fireEvent.change(emailInput, { target: { value: 'test@mail.com' } });
    fireEvent.change(passwordInput, { target: { value: '123' } });
    fireEvent.change(confirmPasswordInput, { target: { value: '123' } });
    const signupButton = await screen.findByText('Sign Up');
    fireEvent.click(signupButton);

    const alert = await screen.findByRole('alert');
    expect(alert).toBeInTheDocument();
    expect(alert).toHaveTextContent('Password is too short (minimum is 6 characters)');
  });
});
