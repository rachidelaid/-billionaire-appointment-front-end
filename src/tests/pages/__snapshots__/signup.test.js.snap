// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page Snapchat should render the signup page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="wrapper"
        >
          <form
            class="card"
          >
            <h1>
              Signup
            </h1>
            <label
              for="name"
            >
              Name
              <input
                id="name"
                placeholder="Name"
                required=""
                type="text"
              />
            </label>
            <label
              for="username"
            >
              Username
              <input
                id="username"
                placeholder="Username"
                required=""
                type="text"
              />
            </label>
            <label
              for="email"
            >
              Email
              <input
                id="email"
                placeholder="Email"
                required=""
                type="email"
              />
            </label>
            <label
              for="password"
            >
              Password
              <input
                id="password"
                min="6"
                placeholder="Password"
                required=""
                type="password"
              />
            </label>
            <label
              for="confirmPassword"
            >
              Confirm Password
              <input
                id="confirmPassword"
                min="6"
                placeholder="Confirm Password"
                required=""
                type="password"
              />
            </label>
            <button
              class="btn"
              type="submit"
            >
              Sign Up
            </button>
          </form>
          <small
            class="option"
          >
            Already have an account?
             
            <a
              href="/login"
            >
              Sign In
            </a>
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="wrapper"
      >
        <form
          class="card"
        >
          <h1>
            Signup
          </h1>
          <label
            for="name"
          >
            Name
            <input
              id="name"
              placeholder="Name"
              required=""
              type="text"
            />
          </label>
          <label
            for="username"
          >
            Username
            <input
              id="username"
              placeholder="Username"
              required=""
              type="text"
            />
          </label>
          <label
            for="email"
          >
            Email
            <input
              id="email"
              placeholder="Email"
              required=""
              type="email"
            />
          </label>
          <label
            for="password"
          >
            Password
            <input
              id="password"
              min="6"
              placeholder="Password"
              required=""
              type="password"
            />
          </label>
          <label
            for="confirmPassword"
          >
            Confirm Password
            <input
              id="confirmPassword"
              min="6"
              placeholder="Confirm Password"
              required=""
              type="password"
            />
          </label>
          <button
            class="btn"
            type="submit"
          >
            Sign Up
          </button>
        </form>
        <small
          class="option"
        >
          Already have an account?
           
          <a
            href="/login"
          >
            Sign In
          </a>
        </small>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
